nengo/processes.py

```diff
@@ -306,7 +306,7 @@ def make_step(self, size_in, size_out, dt, rng):
         presentation_time = float(self.presentation_time)
 
         def step_presentinput(t):
-            i = int(t / presentation_time + 1e-7)
+            i = int((t-dt) / presentation_time + 1e-7)
```

Comment:
One problem is what you need to add to get to the next floating point number depends on the current floating point number. You can figure it out exactly with `lambda x: np.nextafter(x, np.inf) - x`. `f(1) == np.finfo(float).eps == 2.22e-16` (this must be where they get it from), but `f(10) == 1.77e-15`, etc.